### PR TITLE
Build docker image for privacy center

### LINKED ISSUE #1094: Publish a standard Fides Privacy Center image to simplify deployment
### Description
As we are now moving towards adding more features to the privacy center (e.g. #1008, #1009) we also need to improve the quality of the deployment experience for it.

Today we largely leave users to figure out for themselves how to deploy the privacy center. Our documentation page doesn't describe how to deploy it, and largely seems to be written assuming the audience is a developer in the source code itself: https://ethyca.github.io/fidesops/ui/privacy_center/

Ultimately, there are two main ways one could deploy & configure the privacy center:
1. A developer could clone the source code, use `npm install && npm run build` to configure it to your liking, and then use a service like Netlify or Vercel to deploy to production (https://nextjs.org/docs/deployment)
2. Ethyca could publish a Docker image a developer could pull that image and provide overrides for `config.json` and `config.css`, and they could deploy it to any Docker host similar to our other images

I think both options are pretty reasonable, but today we support neither well because our docs aren't clear enough to help with option 1, and we don't publish an image for option 2.

### Assumptions
* Most of our users would prefer a simple, out-of-the-box deployment option for the Privacy Center just using a standard Docker image
* Most of our users would be comfortable configuring the standard image by overriding the config.json and config.css files
* We will eventually support & document both options

### Acceptance Criteria
* MUST publish a standard Fides Privacy Center image with every release, e.g. `ethyca/fidesops:1.8.0-privacy-center`
* MUST be a "production" build of the privacy center image
* MUST support overriding the `config.json` file using Docker (e.g. a volume mount, like `-v ./config:/privacy-center/config` folder or similar)
* MUST support overriding the `config.css` file using Docker (e.g. a volume mount, like `-v ./config:/privacy-center/config` 
* MUST update our deployment guide docs to include the Privacy Center: https://ethyca.github.io/fidesops/deployment/

### Implementation Notes/Details
* See the NextJS deployment docs for best practices on creating a Docker image: https://nextjs.org/docs/deployment. Ultimately we can just adapt the [official NextJS Dockerfile](https://github.com/vercel/next.js/blob/canary/examples/with-docker/Dockerfile) 
* Most Docker images tend to be configured with environment variables, but since the most reasonable experience here is to override entire files (`config.json` and `config.css`), I'm suggesting we use volume mounts. For an example of this, see how the `postgres/postgres` image allows you to provide your own `PGDATA` folder: https://github.com/docker-library/docs/blob/master/postgres/README.md#pgdata

### PR DESCRIPTION
**Pull request by sanders41**
# Purpose

Create a docker image for the privacy center that can be published to docker hub.

I will updated docs etc once we are sure this is how we want things to work. For now to test cd to the privacy-center directory and build the container. `docker build -t privacy-center-test .`. Then run the container `docker run --rm -p 3000:3000 privacy-center-test:latest` and navigate the http://localhost:3000 to verify things are running.

Then in some other directory i.e. `~/test` create a `config` folder then inside there create the following files:

- config.css

```css
/* Add global CSS overrides to this file */
:root:root {
  /* Changes the text color to red */
  --chakra-colors-gray-600: #f00;

  /* Changes the background color to blue */
  --chakra-colors-gray-50: #00f;
}
```

- config.json

```json
{
  "title": "Take control of your data",
  "description": "When you use our services, you’re trusting us with your information. We understand this is a big responsibility and work hard to protect your information and put you in control.",
  "fidesops_host_development": "http://localhost:8080/api/v1",
  "fidesops_host_production": "",
  "logo_path": "/logo.svg",
  "actions": [
    {
      "policy_key": "download",
      "icon_path": "/download.svg",
      "title": "Download your data",
      "description": "We will email you a report of the data related to your account.",
      "identity_inputs": {
        "name": "optional",
        "email": "required",
        "phone": "optional"
      }
    },
    {
      "policy_key": "delete",
      "icon_path": "/delete.svg",
      "title": "Delete your data",
      "description": "We will delete all of your account data. This action cannot be undone.",
      "identity_inputs": {
        "name": "optional",
        "email": "required",
        "phone": "optional"
      }
    }
  ]
}
```

Then from within the test directory start the container with a volume mount `docker run --rm -v $(pwd)/config:/app/config -p 3000:3000 privacy-center-test:latest`. Navigate to http://localhost:3000 again and the privacy center should load, but with different colors this time.

# Changes
- Added a Dockerfile in the privacy-center directory
- Added a script to start the privacy center in docker

# Checklist
- [x] Update [`CHANGELOG.md`](https://github.com/ethyca/fidesops/blob/main/CHANGELOG.md) file
  - [x] Merge in main so the most recent `CHANGELOG.md` file is being appended to
  - [x] Add description within the `Unreleased` section in an appropriate category. Add a new category from the list at the top of the file if the needed one isn't already there.
  - [x] Add a link to this PR at the end of the description with the PR number as the text. example: [#1](https://github.com/ethyca/fidesops/pull/1)
- [x] Applicable documentation updated (guides, quickstart, postman collections, tutorial, fidesdemo, [database diagram](https://github.com/ethyca/fidesops/blob/main/docs/fidesops/docs/development/update_erd_diagram.md).
- If docs updated (select one):
  - [x] documentation complete, or draft/outline provided (tag docs-team to complete/review on this branch)
  - [ ] documentation issue created (tag docs-team to complete issue separately)
- [ ] Good unit test/integration test coverage
- [ ] This PR contains a DB migration. If checked, the reviewer should confirm with the author that the [down_revision correctly references the previous migration](https://ethyca.github.io/fidesops/development/contributing_details/#alembic-migrations) before merging
- [ ] The `Run Unsafe PR Checks` label has been applied, and checks have passed, if this PR touches any external services

# Ticket

Fixes #1094

### CODE REVIEW

**Comment by seanpreston** [comment]
Thanks @sanders41 — this approach looks good

File: .github/workflows/publish_to_dockerhub.yml [hunk 1]
```diff
@@ -43,3 +43,30 @@ jobs:
       - name: Push Prod Tags
         if: steps.check-tag.outputs.match == 'true'
         run: nox -s "push(prod)"
+  push-fidesops-privacy-center-image:
```
Comment (sanders41) [inline .github/workflows/publish_to_dockerhub.yml:46]: I followed the Github documentation on this, but don't have any way to actually test it.

File: .github/workflows/publish_to_dockerhub.yml [hunk 2]
```diff
@@ -43,3 +43,30 @@ jobs:
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Login to DockerHub
+        uses: docker/login-action@v2
+        with:
+          username: ${{ env.DOCKER_USER }}  # Needs updating for fidesops privacy center
+          password: ${{ env.DOCKER_TOKEN }}  # Needs updating for fidesops privacy center
```
Comment (sanders41) [inline .github/workflows/publish_to_dockerhub.yml:55]: These will need to be updated for `ethyca/fidesops-privacy-center` on docker hub

**Review by seanpreston — APPROVED** [review]

File: .github/workflows/publish_to_dockerhub.yml [hunk 3]
```diff
@@ -12,7 +12,7 @@ env:
   DOCKER_TOKEN: ${{ secrets.DOCKER_TOKEN }}
 
 jobs:
-  push-image:
+  push-fidesops-image:
```
Comment (seanpreston) [inline .github/workflows/publish_to_dockerhub.yml:15]: As discussed — would be good to move this into a separate workflow so we don't publish the privacy centre on every release if there are no changes.

**Review by seanpreston — APPROVED** [review]
Thanks

**Pull request merged**